cloud/aws/alb/variables.tf

```diff
@@ -0,0 +1,501 @@
+  description = "Aggregation function and group by for No_healthy_instances detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ".sum(by=['LoadBalancer'])"
+}
+
+variable "no_healthy_instances_transformation_function" {
+  description = "Transformation function for No_healthy_instances detector (mean, min, max)"
+  type        = string
+  default     = "min"
+}
+
+variable "no_healthy_instances_transformation_window" {
+  description = "Transformation window for No_healthy_instances detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "5m"
+}
+
+variable "no_healthy_instances_threshold_critical" {
+  description = "Critical threshold for No_healthy_instances detector"
+  type        = number
+  default     = 1
+}
+
+variable "no_healthy_instances_threshold_warning" {
+  description = "Warning threshold for No_healthy_instances detector"
+  type        = number
+  default     = 100
+}
+
+# Latency detectors
+
+variable "latency_disabled" {
+  description = "Disable all alerting rules for latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "latency_disabled_critical" {
+  description = "Disable critical alerting rule for latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "latency_disabled_warning" {
+  description = "Disable warning alerting rule for latency detector"
+  type        = bool
+  default     = null
+}
+
+variable "latency_notifications" {
+  description = "Notification recipients list for every alerting rules of latency detector"
+  type        = list
+  default     = []
+}
+
+variable "latency_notifications_warning" {
+  description = "Notification recipients list for warning alerting rule of latency detector"
+  type        = list
+  default     = []
+}
+
+variable "latency_notifications_critical" {
+  description = "Notification recipients list for critical alerting rule of latency detector"
+  type        = list
+  default     = []
+}
+
+variable "latency_aggregation_function" {
+  description = "Aggregation function and group by for latency detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "latency_transformation_function" {
+  description = "Transformation function for latency detector (mean, min, max)"
+  type        = string
+  default     = "min"
+}
+
+variable "latency_transformation_window" {
+  description = "Transformation window for latency detector (i.e. 5m, 20m, 1h, 1d)"
+  type        = string
+  default     = "5m"
+}
+
+variable "latency_threshold_critical" {
+  description = "Critical threshold for latency detector"
+  type        = number
+  default     = 3
+}
+
+variable "latency_threshold_warning" {
+  description = "Warning threshold for latency detector"
+  type        = number
+  default     = 1
+}
+
+variable "latency_aperiodic_duration" {
+  description = "Duration for the latency block"
+  type        = string
+  default     = "10m"
```

Comment:
will it raise an alert if no data for more than 10m ?

Reply:
No, this will raise and alert if the datapoints we do get in a 10m window are all above the threshold.  Aperiodic means we don't expect a certain number of datapoints in the time window, so instead of all expected datapoints needing to be above the threshold to alert we just need all datapoints we receive in the time window to be above threshold